### DISCUSSION TITLE
Get depency at router level

### DISCUSSION THREAD

**Discussion by jeanlst** [fastapi/fastapi#4479] [Questions] [answered]
If I happen to have a dependency declared at the router level inside `include_router`, how do I get the object returned by it on specific routes of that router?

```python
api_router = APIRouter()

api_router.include_router(
    items_api_router, dependencies=[Depends(get_current_user)]
)
```

**Comment by jsbeckr** [comment, marked as answer]
If you want to have the return value of a dependency in a route, you have to declare it there.

```python
@router.get("/")
def get_stuff(user: User = Depends(get_current_user):
    return user
```
